treeherder/perf/models.py

```diff
@@ -100,6 +128,26 @@ def has_performance_data(self):
             signature_id=self.id,
         ).exists()
 
+    def has_data_with_historical_value(self):
+        repositories = ['autoland', 'mozilla-central']
+        if self.repository.name in repositories:
+            perf_data = list(
+                PerformanceDatum.objects.filter(
+                    repository_id=self.repository_id,  # leverages (repository, signature) compound index
+                    signature_id=self.id,
+                ).order_by("push_timestamp")
+            )
+
+            most_recent_perf_data = perf_data[-1].push_timestamp
+
+            if data_was_kept_for_one_year(most_recent_perf_data):
```

Comment:
We don't need this check. There's a separate strategy that removes data older than one year.